### PR TITLE
e2e: checking RSS memory for daemons

### LINKED ISSUE #20012: Need e2e testing of Kubelet/Docker memory and CPU usage
To replace test/e2e/monitor_resources.go which was removed in #19987 

ref/ #12429 

cc/ @gmarek @dchen1107

### PR DESCRIPTION
**Pull request by yujuhong**
This fixes #20012

### CODE REVIEW

**Comment by yujuhong** [comment]
This should fix #17774 too.

File: test/e2e/kubelet_perf.go [hunk 1]
```diff
@@ -95,15 +95,53 @@ func runResourceTrackingTest(framework *Framework, podsPerNode int, nodeNames se
...
-
-	summary := rm.GetCPUSummary()
-	Logf("%s", rm.FormatCPUSummary(summary))
-	verifyCPULimits(expected, summary)
+	usageSummary, err := rm.GetLatest()
+	Expect(err).NotTo(HaveOccurred())
+	Logf("%s", rm.FormatResourceUsage(usageSummary))
+	// TODO(yujuhong): Set realistic values after gathering enough data.
+	verifyMemoryLimits(resourceUsagePerContainer{
+		"/kubelet":       &containerResourceUsage{MemoryRSSInBytes: 500 * 1024 * 1024},
```
Comment (vishh) [inline test/e2e/kubelet_perf.go]: RSS of 500MB is pretty large. Do you plan on updating this soon? If not, we might not get much value out of this test.
Not for this PR: How about collecting these metrics for a week and then coming with a number?

File: test/e2e/kubelet_stats.go [hunk 2]
```diff
@@ -364,6 +367,7 @@ func computeContainerResourceUsage(name string, oldStats, newStats *cadvisorapi.
 		CPUUsageInCores:         float64(newStats.Cpu.Usage.Total-oldStats.Cpu.Usage.Total) / float64(newStats.Timestamp.Sub(oldStats.Timestamp).Nanoseconds()),
 		MemoryUsageInBytes:      int64(newStats.Memory.Usage),
 		MemoryWorkingSetInBytes: int64(newStats.Memory.WorkingSet),
+		MemoryRSSInBytes:        int64(newStats.Memory.RSS),
```
Comment (vishh) [inline test/e2e/kubelet_stats.go]: Why are we casting to `int64`?

File: test/e2e/kubelet_perf.go (hunk 1 above)
Comment (yujuhong) [inline test/e2e/kubelet_perf.go]: See TODO two lines above

Comment (vishh) [inline test/e2e/kubelet_perf.go]: Just checking if you plan on resolving that TODO anytime soon ;)

File: test/e2e/kubelet_stats.go (hunk 2 above)
Comment (yujuhong) [inline test/e2e/kubelet_stats.go]: Good question. I don't remember. Added a commit to change everything to uint64

File: test/e2e/kubelet_perf.go (hunk 1 above)
Comment (yujuhong) [inline test/e2e/kubelet_perf.go]: Even getting numbers is better than nothing, but yes, I plan to resolve that in a week or so :-)

**Comment by wojtek-t** [comment]
@k8s-bot e2e test this please github issue: #21261

**Pull request merged**